src/main/java/org/apache/commons/lang3/StringUtils.java

```diff
@@ -4196,7 +4196,8 @@ public static String join(final Iterable<?> iterable, final String separator) {
      * @return the joined String.
      * @throws java.lang.IllegalArgumentException if a null varargs is provided
      */
-    public static String joinWith(final String separator, final Object... objects) {
+    @SuppressWarnings("deprecation")
```

Comment:
It's better to keep the scope of this annotation as small as possible. I'd rather assign the result of {{ObjectUtils.toString(iterator.next())}} and add the annotation there.